Modules/Numerics/Optimizersv4/include/itkGradientDescentLineSearchOptimizerv4.hxx

```diff
@@ -28,7 +28,7 @@ GradientDescentLineSearchOptimizerv4Template<
 {
   this->m_MaximumLineSearchIterations = 20;
   this->m_LineSearchIterations = NumericTraits<unsigned int>::ZeroValue();
```

Comment:
I think this is intentionally left out, I guess, `unsigned int{}` doesn't work, `(unsigned int){}` seems to compile, but probably `0U` is even better, IMHO.
P.S. There are not many similar situations (e.g. _unsigned char_, _long double_, etc), some are in itkFreeSurferAsciiMeshIO.cxx, AFAIK.

Reply:
@issakomi Please check the two commits that I added to address your comments. Also please consider approving this pull request 😃